lib/Facebook.js

```diff
@@ -815,9 +815,46 @@ function Facebookbot(configuration) {
         });
     };
 
+    var insights_api = {
+        get_insights: function (metrics, since, until, cb) {
+            var uri = 'https://' + api_host + '/v2.8/me/insights?access_token=' + configuration.access_token;
 
+            if (metrics) {
+                if (typeof(metrics) == 'string') {
+                    uri += '&metric=' + metrics;
+                } else {
+                    uri += '&metric=' + metrics.join(',');
+                }
+            }
 
+            if (since) {
+                uri += '&since=' + since;
+            }
 
+            if (until) {
+                uri += '&until=' + until;
+            }
+
+            request({
+                method: 'GET',
+                json: true,
+                uri: uri
+            }, function (err, res, body) {
+                if (err) {
+                    facebook_botkit.log('Could call Facebook Insights API');
+                    cb(err);
+                } else {
+                    if (body.error) {
+                        facebook_botkit.log('ERROR in Facebook Insights API call: ', body.error.message);
+                        cb(body.error);
```

Comment:
body.error is not an instance of Error, is it?
If it isn't, then it would probably be better to use `cb(new Error(body.error.message));`

Reply:
No it's not, ````body.error```` is the error object returned by FB, it contains other useful data than message.

If we call the callback with an error object and juste the message, we couldn't use those useful data (code, ...).